### PR TITLE
Setting naming convention for exporter packages

### PR DESCRIPTION
**Pull request by toumorokoshi**
There is an implicit convention around exporter packages in several
language instrumentations. Adding the convention to the spec standardizes
the convention.

### CODE REVIEW

File: specification/library-guidelines.md [hunk 1]
```diff
@@ -83,6 +83,9 @@ Vendors are encouraged to keep protocol-specific exporters as simple as possible
 
 End users should be given the flexibility of making many of the decisions regarding the queuing, retrying, tagging, batching functionality that make the most sense for their application. For example, if an application's telemetry data must be delivered to a remote backend that has no guaranteed availability the end user may choose to use a persistent local queue and an `Exporter` to retry sending on failures. As opposed to that for an application that sends telemetry to a locally running Agent daemon, the end user may prefer to have a simpler exporting configuration without retrying or queueing.
 
+If additional exporters for the sdk are provided as separate libraries, the
+name of the library should be prefixed with opentelemetry-exporter. For example, "opentelemetry-exporter-jaeger".
```
Comment (jtescher) [inline specification/library-guidelines.md:87]: Could `opentelemetry-jaeger` be acceptable here as well? Or is there a case where specifying `exporter` as a disambiguation is required?

File: specification/glossary.md [hunk 2]
```diff
@@ -12,7 +12,13 @@ Denotes the library that implements the *OpenTelemetry API*.
 See [Library Guidelines](library-guidelines.md#sdk-implementation) and
 [Library resource semantic conventions](resource/semantic_conventions/README.md#telemetry-sdk)
 
-<a name="instrumented_library"></a>
+<a name="telemetry_sdk"></a>
```
Comment (SergeyKanzhelev) [inline specification/glossary.md:15]: change the link below:

> `Denotes the library that provides the instrumentation for a given [Instrumented Library](#instrumented_library).`

File: specification/library-guidelines.md (hunk 1 above)
Comment (SergeyKanzhelev) [inline specification/library-guidelines.md:87]: are dashes a requirement? It may not be aligned with the language naming convention. 

If not maybe rephrase it as "prefix must include names OpenTelemetry and Exporter"

Comment (carlosalberto) [reply specification/library-guidelines.md:87]: Another case is having more than one package for vendor specific stuff, e.g. in OTel Java, besides the Jaeger Exporter, we have a [Jaeger Remote Sampler](https://github.com/open-telemetry/opentelemetry-java/tree/master/sdk_contrib/jaeger_remote_sampler), so it will be handy so be a little explicit on the name of each artifact.

Comment (jtescher) [reply specification/library-guidelines.md:87]: Are packages allowed to contain both remote samplers and exporters?

File: specification/library-guidelines.md [hunk 3]
```diff
@@ -83,6 +83,9 @@ Vendors are encouraged to keep protocol-specific exporters as simple as possible
 
 End users should be given the flexibility of making many of the decisions regarding the queuing, retrying, tagging, batching functionality that make the most sense for their application. For example, if an application's telemetry data must be delivered to a remote backend that has no guaranteed availability the end user may choose to use a persistent local queue and an `Exporter` to retry sending on failures. As opposed to that for an application that sends telemetry to a locally running Agent daemon, the end user may prefer to have a simpler exporting configuration without retrying or queueing.
 
+If additional exporters for the sdk are provided as separate libraries, the
```
Comment (arminru) [inline specification/library-guidelines.md:86]:
```suggestion
If additional exporters for the OpenTelemetry SDK are provided as separate libraries, the
```

File: specification/library-guidelines.md (hunk 1 above)
Comment (arminru) [inline specification/library-guidelines.md:87]:
```suggestion
name of the library should be prefixed with the terms "OpenTelemetry" and "Exporter" in accordance with the naming conventions of the respective technology.
An exporter for Jaeger in Java would, for example, be called `opentelemetry-exporter-jaeger`.
```

**Comment by bogdandrutu** [comment]
@toumorokoshi please rebase and fix all comments :)

File: specification/library-guidelines.md (hunk 1 above)
Comment (toumorokoshi) [reply specification/library-guidelines.md]: i'll follow up on the general convention in a separate PR.

**Comment by toumorokoshi** [comment]
@bogdandrutu done!
